### DISCUSSION TITLE
@ckeditor/ckeditor5-engine: detect for('upcast') and for('downcast')?

### DISCUSSION THREAD

**Discussion by neongreen** [DefinitelyTyped/DefinitelyTyped#54509] [Issues with a @types package] [answered]
[`Conversion`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ckeditor__ckeditor5-engine/v27/src/conversion/conversion.d.ts) currently features:

```
for(groupName: string): DowncastHelpers | UpcastHelpers;
```

I don't know TypeScript well, but is it possible to provide a pair of signatures instead?

```
for(groupName: 'upcast'): UpcastHelpers;
for(groupName: 'downcast'): DowncastHelpers;
```

cc @fedemp

**Comment by neongreen** [comment, marked as answer]
Well, I guess we'd also have to cover `dataDowncast` and `editingDowncast`.

**Reply by fedemp** [reply]
`upcast`, `datadowncast`, and `editingdowncast` are already in `master`. I'll create now a PR for `downcast`.

**Reply by fedemp** [reply]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54553

**Reply by fedemp** [reply]
@neongreen Could you mark this as answered if the issue is resolved? Just so we know what needs to be fixed. :slightly_smiling_face:
